File: Java/src/classes/Carte.java
package src.classes;
public class Carte {

    private int valeur;
    
    public Carte(int valeur) {
        this.valeur = valeur;
    }

    public int getValeur() {
        return valeur;
    }

    @Override
    public boolean equals(Object autre) {
        if (this == autre) {
            return true;
        }

        if (autre == null || getClass() != autre.getClass()) {
            return false;
        }

        Carte other = (Carte) autre;

        return this.valeur == other.valeur;
    }

    @Override
    public String toString() {
        return "Carte " + this.valeur;
    }
}

File: C++/src/classes/Carte.cpp
#include "Carte.h"
#include <string>

Carte::Carte(int valeur) : valeur(valeur) {}

std::string Carte::toString() const {
    return "Carte " + std::to_string(valeur);
}

bool Carte::operator==(const Carte& autre) const {
    return this->valeur == autre.valeur;
}

int Carte::getValeur() const {
    return valeur;
}
